components/support/migration/src/main/java/mozilla/components/support/migration/AbstractMigrationProgressActivity.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.migration
+
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.migration.state.MigrationProgress
+import mozilla.components.support.migration.state.MigrationStore
+
+/**
+ * An activity that notifies on migration progress. Should be used in tandem with [MigrationIntentProcessor].
+ */
+abstract class AbstractMigrationProgressActivity : AppCompatActivity(), MigrationStateListener {
```

Comment:
I am not sure if this is strictly needed. It looks like this abstract activity mostly maps the state in the store to another abstraction layer. So I guess the implementing activity could just subscribe to the store itself and we would not need more code and more abstraction? Anyhow, this is still okay to land, I think - especially given the time constraints. :)

Reply:
Yeah I felt the same way. I only opted for this so that have control of the `onBackPressed` callback to make it `final` and do nothing. Without it, that would be a big loop-hole that apps need to remember to do.